hlapipeline/alignimages.py

```diff
@@ -616,30 +633,123 @@ def update_image_wcs_info(tweakwcs_output,imagelist):
             image_name = imagelist[imgctr]
             if imgctr > 0: #close previously opened image
-                print("CLOSE {}".format(hdulist[0].header['FILENAME'])) #TODO: Remove before deployment
+                imageName = hdulist[0].header['FILENAME']
+                print("CLOSE {}\n".format(imageName)) #TODO: Remove before deployment
                 hdulist.flush()
                 hdulist.close()
+
+                # Create headerlet
+                out_headerlet = headerlet.create_headerlet(imageName, hdrname=wcsName, wcsname=wcsName)
+
+                # Update headerlet
+                update_headerlet_phdu(item, out_headerlet)
+
+                if imageName.endswith("flc.fits"):
+                    headerlet_filename = imageName.replace("flc", "flt_hlet")
+                if imageName.endswith("flt.fits"):
+                    headerlet_filename = imageName.replace("flt", "flt_hlet")
+                out_headerlet.writeto(headerlet_filename,clobber=True)
+                print("Wrote headerlet file {}.\n\n".format(headerlet_filename))
+                out_headerlet_dict[imageName] = headerlet_filename
             hdulist = fits.open(image_name, mode='update')
+            #generate wcs name for updated image header, headerlet
+            if not hdulist['SCI',1].header['WCSNAME'] or hdulist['SCI',1].header['WCSNAME'] =="": #Just in case header value 'wcsname' is empty.
+                wcsName = "FIT_{}".format(item.meta['catalog_name'])
+            else:
+                wname = hdulist['sci', 1].header['wcsname']
+                if "-" in wname:
+                    wcsName = '{}-FIT_{}'.format(wname[:wname.index('-')], item.meta['tweakwcs_info']['catalog'])
+                else:
+                    wcsName = '{}-FIT_{}'.format(wname, item.meta['tweakwcs_info']['catalog'])
+
             sciExtDict = {}
             for sciExtCtr in range(1, amutils.countExtn(hdulist) + 1): #establish correct mapping to the science extensions
                 sciExtDict["{}".format(sciExtCtr)] = fileutil.findExtname(hdulist,'sci',extver=sciExtCtr)
             imgctr += 1
-        updatehdr.update_wcs(hdulist, sciExtDict["{}".format(item.meta['chip'])], item.wcs, wcsname='TWEAKDEV', reusename=True, verbose=True) #TODO: May want to settle on a better name for 'wcsname'
+        updatehdr.update_wcs(hdulist, sciExtDict["{}".format(item.meta['chip'])], item.wcs, wcsname=wcsName, reusename=True, verbose=True)
         print()
-    print("CLOSE {}".format(hdulist[0].header['FILENAME'])) #TODO: Remove before deployment
+    imageName = hdulist[0].header['FILENAME']
+    print("CLOSE {}\n".format(imageName)) #TODO: Remove before deployment
     hdulist.flush() #close last image
     hdulist.close()
 
+    # Create headerlet
+    out_headerlet = headerlet.create_headerlet(imageName, hdrname=wcsName, wcsname=wcsName)
+
+    # Update headerlet
```

Comment:
Sorry, I am being dense.  Why is this second update needed.  I know that if both FLT and FLC images exit, we need to update the headers of both, but this does not seem to be the issue.  Just a few words of explanation to me would be super.